numpy/lib/arraysetops.py

```diff
@@ -551,6 +551,50 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
+    # Convert booleans to uint8 so we can use the fast integer algorithm
+    if ar1.dtype == np.bool_:
+        ar1 = ar1.view(np.uint8)
+    if ar2.dtype == np.bool_:
+        ar2 = ar2.view(np.uint8)
+
+    # Check if we can use a fast integer algorithm:
+    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) and
+                      np.issubdtype(ar2.dtype, np.integer))
+
+    # Make sure we won't encounter memory errors
+    if integer_arrays:
+        ar2_min = np.min(ar2)
+        ar2_max = np.max(ar2)
+        ar2_range = ar2_max - ar2_min
```

Comment:
This can overflow if `ar2_min == np.int8(-100)`, `ar2_max == np.int8(100)`. Does the algorithm work anyway?

Reply:
It seems it actually works in this case but it does spit out warnings:
```
>>> import numpy as np
>>> y = np.array([-100, 5, 100]).astype(np.int8)
>>> x = np.array([5, 50, 100]).astype(np.int8)
>>> np.isin(x, y)
array([ True, False,  True])
```

With warning:
```
/opt/conda/lib/python3.7/site-packages/numpy-1.17.0.dev0+5d12e37-py3.7-linux-x86_64.egg/numpy/lib/arraysetops.py:568: RuntimeWarning: overflow encountered in byte_scalars
  ar2_range = ar2_max - ar2_min
/opt/conda/lib/python3.7/site-packages/numpy-1.17.0.dev0+5d12e37-py3.7-linux-x86_64.egg/numpy/lib/arraysetops.py:576: RuntimeWarning: invalid value encountered in log10
  ((np.log10(ar2_range + 1) - 2.27) / 0.927)
```

It looks like the optimal_parameters flag catches this invalid value and the old algorithm is used instead. Should I check for this or leave it as is?

Reply:
I check for overflow explicitly in 40ad85e. Let me know what you think.